server/src/pages/homepage/controller.js

```diff
@@ -5,9 +5,26 @@ let model;
 
 exports.launch = function(m) {
   model = m;
+  if (window.wantsauth) {
+    if (window.wantsauth.getAuthData()) {
+      updateModel(window.wantsauth.getAuthData());
+    } else {
+      window.wantsauth.addAuthDataListener((data) => {
+        updateModel(data);
+        render();
+      });
+    }
+  }
   render();
 };
 
+function updateModel(authData) {
+  Object.assign(model, authData);
+  if (!model.hasFxa) {
```

Comment:
Since this function is only called on `wantsauth`, I don't think we need this `if`.  We want to set or update `hasFxa` upon receiving auth data.

Reply:
+1